reactioncommerce/reaction
# (fix): Add permission checks to template method and publication

Adds permission checks to the `template/email/update` server method.

I've tested and verified that as an admin you can still view and modify email templates.
I've also tested and verified that users without the `reaction-templates` role for the primary shop cannot view or subscribe to the templates, and cannot modify them.

We currently _do not_ use shop specific templates, the default marketplace install of Reaction disables template editing for shops and uses the primary shop's templates for all emails. I've added this code to be compatible with marketplace shops, but it's not enabled right now anyway, and there's not a great way to test that.

**To test:**
1. Verify that as the admin user or user with the `reaction-templates` role you can still view and modify templates.
2. As an unauthenticated user. (incognito window works)
3. Attempt to subscribe to the templates publication, verify that you are not sent any template documents.
4. Attempt to modify a template by calling the `template/email/update` method from the client console. Verify that you are not able to do so successfully.

![image](https://user-images.githubusercontent.com/1203639/35468148-98f311dc-02d5-11e8-8ed4-d631bdf3afc0.png)
